src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1216,12 +1246,49 @@ export class StudyViewPageStore {
     }
 
     @autobind
-    @action resetGeneFilter() {
+    @action resetMutatedGeneFilter() {
         if(this._mutatedGeneFilter.length > 0) {
             this._mutatedGeneFilter = [];
         }
     }
 
+    @autobind
+    @action
+    addFusionGeneFilters(uniqueKeys: string[]) {
+        const genes = uniqueKeys.map(uniqueKey => parseMutationUniqueKey(uniqueKey));
+
+        trackStudyViewFilterEvent("geneFilter", this);
+
+        genes.forEach(gene => this.geneMapCache[gene.entrezGeneId] = gene.hugoGeneSymbol);
```

Comment:
i think "genemapCache" name could be improved.  Looks like it's entrezGeneId to hugoGeneSymbol?  why is that a cache?

Reply:
agree. it is just a normal map.

Reply:
@alisman yeah, we were trying to emphasize the purpose of the variable

Reply:
this won't be needed anymore once hugoGeneSymbols are supported. removed in my pr https://github.com/cBioPortal/cbioportal-frontend/pull/2714/files#diff-a3ffe2439b6c26eb3cbf8050e37410a8L703

Reply:
@kalletlak @alisman  just in case I don't introduce more conflicts. I will leave the name as it.